### PR TITLE
Allow the RPC Subscribe method to see all events

### PR DESCRIPTION
**Pull request by silasdavis**
Those events from both consensus and application manager.

Doesn't fix tests. Because Ben.

Sorry, I have caught some bullshit formatting changes here.

manager/eris-mint/pipe.go is the real entry point here

### CODE REVIEW

**Comment by silasdavis** [comment]
I've split this commit to remove some of the incidental cruft

File: event/events.go [hunk 1]
```diff
@@ -30,29 +30,61 @@ import (
...
-	return true, nil
+	return nil
+}
+
+type multiplexedEvents struct {
+	eventEmitters []EventEmitter
+}
+
+// Subscribe to an event.
+func (multiEvents *multiplexedEvents) Subscribe(subId, event string, callback func(evts.EventData)) error {
```
Comment (benjaminbollen) [inline event/events.go]: Please raise an issue for adding unit tests for this event multiplexing.

**Comment by benjaminbollen** [comment]
also correct the signature ```
# github.com/eris-ltd/eris-db/test/mock
test/mock/pipe.go:45: cannot use eventer (type *eventer) as type event.EventEmitter in field value:
	*eventer does not implement event.EventEmitter (wrong type for Subscribe method)
		have Subscribe(string, string, func(events.EventData)) (bool, error)
		want Subscribe(string, string, func(events.EventData)) error
```

**Comment by benjaminbollen** [comment]
and
``` github.com/eris-ltd/eris-db/event
event/event_cache_test.go:79: cannot use mee (type *mockEventEmitter) as type EventEmitter in argument to NewEventSubscriptions:
	*mockEventEmitter does not implement EventEmitter (wrong type for Subscribe method)
		have Subscribe(string, string, func(events.EventData)) (bool, error)
		want Subscribe(string, string, func(events.EventData)) error
event/event_cache_test.go:124: cannot use mee (type *mockEventEmitter) as type EventEmitter in argument to NewEventSubscriptions:
	*mockEventEmitter does not implement EventEmitter (wrong type for Subscribe method)
		have Subscribe(string, string, func(events.EventData)) (bool, error)
		want Subscribe(string, string, func(events.EventData)) error
event/event_cache_test.go:173: cannot use mee (type *mockEventEmitter) as type EventEmitter in argument to NewEventSubscriptions:
	*mockEventEmitter does not implement EventEmitter (wrong type for Subscribe method)
		have Subscribe(string, string, func(events.EventData)) (bool, error)
		want Subscribe(string, string, func(events.EventData)) error
FAIL	github.com/eris-ltd/eris-db/event [build failed]```

**Comment by benjaminbollen** [comment]
and trivially ```github.com/eris-ltd/eris-db/rpc/tendermint/client
rpc/tendermint/client/client.go:4: imported and not used: "fmt"```
is causing these tests to not show up in the circle output

**Comment by benjaminbollen** [comment]
LGTM

**Pull request merged**